Lib/test/test_grammar.py

```diff
@@ -301,6 +302,12 @@ def one():
         [list][0]: type
         my_lst[one()-1]: int = 5
         self.assertEqual(my_lst, [5])
+        exec(textwrap.dedent("""
+            def f():
+                global x
+                x = 1
+            x:int = 3
+            """))
```

Comment:
I would reserve this test file only for basic things. I propose to add this test to either `test_symtable.py` or `test_syntax.py`.

Also there is even a simpler repro for the issue:
```python
global x
x: int  # SyntaxError
```
plus I would add test cases for `nonlocal` as well (even though it already works).

Reply:
It seems that `test_syntax` tests all raise SyntaxErrors, but in this case, we want to test that the solution is valid. On the other hand, `test_symtable` seems to test the symtable properties directly, as opposed to check that something is a valid Syntax or not. I am a bit unsure on which file to put the reproducer (along with the "nonlocal" one).

Reply:
I have openend https://github.com/python/cpython/pull/9872 as there are no current way to test for nonlocal declarations in the symtable python module.